src/ray/object_manager/pull_manager.cc

```diff
@@ -170,11 +173,16 @@ void PullManager::UpdatePullsBasedOnAvailableMemory(size_t num_bytes_available)
     RAY_CHECK(last_request_it != pull_request_bundles_.end());
     DeactivatePullBundleRequest(last_request_it, &object_ids_to_cancel);
   }
+  for (const auto &obj_id : object_ids_to_cancel) {
+    // Call the cancellation callback outside of the lock.
+    cancel_pull_request_(obj_id);
+  }
 
   TriggerOutOfMemoryHandlingIfNeeded();
 
   for (const auto &obj_id : objects_to_pull) {
     if (object_ids_to_cancel.count(obj_id) == 0) {
+      absl::MutexLock lock(&active_objects_mu_);
```

Comment:
This will acquire a lock for each iteration right? Why don't we move the loc outside the for loop?